wasm-runtime/polkadot/src/codec/slicable.rs

```diff
@@ -34,7 +36,7 @@ pub trait Slicable: Sized {
 	fn to_vec(&self) -> Vec<u8> {
 		self.as_slice_then(|s| s.to_vec())
 	}
-	fn set_as_slice<F: FnOnce(&mut [u8]) -> bool>(set_slice: F) -> Option<Self>;
+	fn set_as_slice<F: Fn(&mut [u8], usize) -> bool>(set_slice: &F) -> Option<Self>;
```

Comment:
It's more restrictive to take `&F` than `F` here because `Fn` is implemented for `&F` already.

Reply:
see later comment.